src/lib.rs

```diff
@@ -941,6 +942,11 @@ pub fn delegate(tokens: TokenStream) -> TokenStream {
                         }
                         get_const(#expr)
                     }}
+                } else if let Some(expr_attr) = expr_attr {
```

Comment:
I think that `$expr` should be applied as the very last transformation, and it should expand to whatever would otherwise be generated. So here:
```rust
delegate::delegate! {
            to self.0 {
                #[expr($expr.unwrap())] // error when we have ($
                fn baz(self) -> u32;
            }
        }
```
it should expand to `self.0.baz().unwrap()`, not `self.0.unwrap()`. Otherwise users would have to reimplement the existing generics, return type, etc. logic inside the attribute.